spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/JobsController.java

```diff
@@ -72,9 +78,44 @@ public void save(@RequestParam("name") String name, @RequestParam("definition")
 		if (distributedJobLocator.getJobNames().contains(name)) {
 			throw new BatchJobAlreadyExistsException(name);
 		}
+		if(ComposedJobUtil.isComposedJobDefinition(definition)){
+			moduleDefinitionService.compose(ComposedJobUtil.getComposedJobModuleName(name), ModuleType.job,definition, false);
+		}
 		super.save(name, definition, deploy);
 	}
 
+
+	/**
+	 * Request removal of an existing resource definition (stream or job).
+	 *
+	 * @param name the name of an existing definition (required)
+	 */
+	@RequestMapping(value = "/definitions/{name}", method = RequestMethod.DELETE)
+	@ResponseStatus(HttpStatus.OK)
+	public void delete(@PathVariable("name") String name) throws Exception {
+		super.delete(name);
+		if( moduleDefinitionService.findDefinition(
+				ComposedJobUtil.getComposedJobModuleName(name), ModuleType.job) != null) {
+			moduleDefinitionService.delete(ComposedJobUtil.getComposedJobModuleName(name), ModuleType.job);
+		}
+	}
+
+	/**
+	 * Request removal of all definitions.
+	 */
+	@RequestMapping(value = "/definitions", method = RequestMethod.DELETE)
+	@ResponseStatus(HttpStatus.OK)
+	public void deleteAll() throws Exception {
```

Comment:
What is the use case here?  Deleting all composed job definitions seems like an odd option to me unless I'm missing something...